### ISSUE TITLE
[ttx] ttx should bump GSUB version to 1.1 when compiling GSUB FeatureVariations

### ISSUE THREAD

**Issue by Lorp** [fonttools/fonttools#1215]
If you add <FeatureVariations> XML to an existing Version 0x00010000 GSUB table, then ttx ignores that section without a warning, even in verbose mode. You must manually edit the Version to 0x00010001 for it to compile correctly. It would be useful if ttx automatically bumped the Version if FeatureVariations are detected.

**Comment by justvanrossum** [comment]
I was bit by this when adding FeatureVariations programmatically. It's probably a good idea to do this. I'll have a quick look.

**Comment by justvanrossum** [comment]
Hmm, due to the data-driven nature of how these tables are implemented I don't immediately see how to do an automatic version bump. Any ideas, @behdad?

Perhaps it's easier to issue a warning (or even an error) if FeatureVariations is present but the table version doesn't match.

**Comment by khaledhosny** [comment]
This can happen with other tables as well (e.g. the various versions of `OS/2` table) and IIRC correctly, the consensus was that the user has to take care of setting the right version number.

**Comment by behdad** [comment]
> This can happen with other tables as well (e.g. the various versions of OS/2 table) and IIRC correctly, the consensus was that the user has to take care of setting the right version number.

Yes, this is not the right level to do such magic. This applies to all kinds of tables.

Better to implement varLib support for FeatureVariations such that people don't have to manually generate these.

**Comment by Lorp** [comment]
Thanks for all the comments. I understand why silently updating version fields might not be a good idea. Still, it seems like this kind of thing (including `OS/2` edits that require a version bump) would be a great candidate for warnings. I’ve done some productive `gvar` hacking this way, and it seems the kind of hacking that’s possible via editing TTX sources is unclear.

**Comment by behdad** [comment]
I think I can generate a warning easily.

**Comment by behdad** [comment]
Here's a sketch. Someone finish it:
```diff
diff --git a/Lib/fontTools/ttLib/tables/otBase.py b/Lib/fontTools/ttLib/tables/otBase.py
index 18954e1c..8295a9e7 100644
--- a/Lib/fontTools/ttLib/tables/otBase.py
+++ b/Lib/fontTools/ttLib/tables/otBase.py
@@ -695,6 +695,8 @@ class BaseTable(object):
                                writer['LookupType'] = ref
                        else:
                                if conv.aux and not eval(conv.aux, None, table):
+                                       if value is not None:
+                                               log.warning(...conv.name...ignored...)
                                        continue
                                try:
                                        conv.write(writer, font, table, value)
```
